src/rs232.rs

```diff
@@ -110,7 +110,8 @@ impl RS232 {
 
         for _ in 1..3 {
             if let Ok(_) = self.read_line(&mut buffer) {
-                if buffer.len() >= 15 && buffer.starts_with("$G") &&
+                if buffer.len() >= 15 &&
+                    buffer.chars().nth(0).unwrap() == '$' &&
```

Comment:
A few things to say here:

* It sucks a bit that we have less things to check now but what can you do. So good catch.
* I did not realized that in the last patch you introduced an 'unwrap()' call. 'unwrap()' should be avoided and when used, a comment briefly explaining why it's justified in this case would be good. If you keep the unwrap(), you could add the comment in a separate patch on top of this. If you decide to remove these calls, the patch to remove it, should go before this one.
* The commit log description part:
   * Each line is supposed to be 75 chars max.
   * IMO "test" -> "check" is better.

Reply:
> It sucks a bit that we have less things to check now but what can you do. So good catch.

Other than actually parsing a full NMEA sentence and verifying it’s checksum? Not sure. This check can cause false positives, but it will also work with every device producing NMEA. I think it’s better than it works with more devices in more markets than guarding against it sometimes picking the wrong device.

Could use a regex like `\$[A-Z]{5},` but that would be a lot slower depending on the number of connected devices.

> I did not realized that in the last patch you introduced an 'unwrap()' call. 'unwrap()' should be avoided and when used, a comment briefly explaining why it's justified in this case would be good. If you keep the unwrap(), you could add the comment in a separate patch on top of this. If you decide to remove these calls, the patch to remove it, should go before this one.

Character 0 and 6 can’t be None as the buffer length is guaranteed to be 15+ characters. (Right?) Are there any other issues with unwrap() than guarding against None panics? It seemed like the simplest tool to get the job done.

> The commit log description part. Each line is supposed to be 75 chars max.

Amended.